src/cms/views/dashboard/dashboard_view.py

```diff
@@ -67,18 +68,33 @@ def get(self, request, *args, **kwargs):
 
         # RSS FEED
         language_slug = translation.get_language()
-        feed = feedparser.parse(RSS_FEED_URLS[language_slug])
-        # select five most recent feeds
-        feed["entries"] = feed["entries"][:3]
-        # decode html entities like dash and split after line break
-        for entry in feed["entries"]:
-            entry["summary"] = html.unescape(entry["summary"]).split("\n")[0]
+        has_timed_out = False
+        feed = []
+        rss_feed_url = RSS_FEED_URLS[language_slug]
+
+        try:
+            requests.get(rss_feed_url, timeout=5.0)
```

Comment:
This will do one additional request just to check whether the feed is available... This will make the process slower each time the feed is actually available (which is 99.9% of the time), so I don't think it's a good idea. It would be better to actually use the response of the first request to directly parse the RSS feed:
```suggestion
            response = requests.get(rss_feed_url, timeout=5)
            feed = feedparser.parse(response.text)
```

Reply:
Thanks for this hint!